src/SolidStateDetector/Semiconductor.jl

```diff
@@ -89,23 +92,33 @@ function Semiconductor{T}(dict::AbstractDict, input_units::NamedTuple, outer_tra
     transformations = combine_transformations(inner_transformations, outer_transformations)
     geometry = Geometry(T, dict["geometry"], input_units, transformations)
 
-    charge_trapping_model = if haskey(dict, "charge_trapping_model") &&
-                               haskey(dict["charge_trapping_model"], "model") &&
-                               dict["charge_trapping_model"]["model"] == "Boggs"
-        BoggsChargeTrappingModel{T}(dict["charge_trapping_model"], temperature = temperature)
-    elseif haskey(dict, "charge_trapping_model") &&
-                               haskey(dict["charge_trapping_model"], "model") &&
-                               dict["charge_trapping_model"]["model"] == "ConstantLifetime"
-        constant_lifetime_ctm_dict = deepcopy(dict["charge_trapping_model"])
-        if haskey(constant_lifetime_ctm_dict, "inactive_layer_geometry")
-            constant_lifetime_ctm_dict["inactive_layer_geometry"] = Geometry(T, dict["charge_trapping_model"]["inactive_layer_geometry"], input_units, transformations)
-        end
+
+    constant_lifetime_ctm_dict = haskey(dict, "charge_trapping_model") ? deepcopy(dict["charge_trapping_model"]) : Dict{String, Any}()
+
+    if haskey(constant_lifetime_ctm_dict, "inactive_layer_geometry")
+        constant_lifetime_ctm_dict["inactive_layer_geometry"] = Geometry(T, constant_lifetime_ctm_dict["inactive_layer_geometry"], input_units, transformations)
+    end
+
+    charge_trapping_model_inactive = if haskey(constant_lifetime_ctm_dict, "model_inactive") &&
```

Comment:
Somewhere here: if it has the key `model_inactive` and `model`, create a combined one from the two.